frontend/src/pages/PipelineDetails.tsx

```diff
@@ -59,6 +61,14 @@ interface PipelineDetailsState {
   versions: ApiPipelineVersion[];
 }
 
+type Origin = {
+  isRecurring: boolean;
```

Comment:
Correct! The `isRecurring` is not necessary; however, the inline expression will become a little weird like `id = runId ? runId : recurringRunId`. Thus, we choose to add a flag here, which is consistent to previous implementation in `NewRunV2`

Reply:
I personally don't feel `id = runId ? runId : recurringRunId` being weird. It can be also written as 
```javascript
id = rundId || recurringRunId;
``` 
which might look cleaner?

Anyway, this is a nitpick, feel free to ignore if you prefer a dedicated flag.